Model not provided to error substate templates

As per [the Ember guides](https://guides.emberjs.com/release/routing/loading-and-error-substates/), the thrown exception of promise rejection reason should be provided as the `model`.

However, glint rejects these templates accessing `{{@model}}`:

```
app/templates/error.hbs:11:6 - error TS2339: Property 'model' does not exist on type 'EmptyObject'.

11   {{@model}}
        ~~~~~
```

--- a/docs/ember/routes-and-controllers.md
+++ b/docs/ember/routes-and-controllers.md
@@ -34,3 +34,16 @@ export default class MyRoute extends Route {
 {{this.model}} {{! MyModelType }}
 {{@model}} {{! MyModelType }}
 ```
+
+For `error` substate routes, the type of `{{@model}}` will not be automatically inferred. You will need to create a backing class for the route if you consume its model in the corresponding template:
+
+```typescript
+export default class ErrorRoute extends Route<Error> {
+  // ...
+}
+```
+
+<!-- prettier-ignore -->
+```handlebars
+{{@model}} {{! Error }}
+```

